src/src/com/microsoft/aad/adal/SSOStateContainer.java

```diff
@@ -0,0 +1,94 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.annotations.SerializedName;
+
+class SSOStateContainer {
+    /**
+     * the version number of {@link SSOStateContainer}
+     */
+    @SerializedName("version")
+    private final int version = 1;
+
+    /**
+     * the {@link SSOStateContainer} stores the FRT tokenCacheItem of the given
+     * user
+     */
+    @SerializedName("tokenCacheItems")
+    private final List<SerializedTokenCacheItem> tokenCacheItems = new ArrayList<SerializedTokenCacheItem>();
+
+    /**
+     * Check if the cache item contains family refresh token if true, value the
+     * tokenItem with this cache item
+     * 
+     * @param item
+     */
+    SSOStateContainer(final TokenCacheItem tokenItem) {
+        // only FRT is stored here
+        if (tokenItem == null) {
+            throw new IllegalArgumentException("tokenItem is null");
+        }
+
+        if (StringExtensions.IsNullOrBlank(tokenItem.getFamilyClientId())) {
+            throw new IllegalArgumentException("tokenItem does not contain family refresh token");
+        }
+
+        final SerializedTokenCacheItem serializedTokenCacheItem = new SerializedTokenCacheItem(tokenItem);
+        this.tokenCacheItems.add(serializedTokenCacheItem);
+    }
+
+    /**
+     * return the token cache item in this blob container object
+     * 
+     * @return tokenCacheItem
+     * @throws AuthenticationException
+     */
+    private TokenCacheItem getTokenItem() throws AuthenticationException {
+        if (tokenCacheItems == null || tokenCacheItems.isEmpty()) {
+            throw new AuthenticationException(ADALError.FAMILY_REFRESH_TOKEN_NOT_FOUND,
+                    "There is no family refresh token cache item in the SSOStateContainer.");
+        }
+        return tokenCacheItems.get(0).toTokenCacheItem();
+    }
+
+    String serialize() {
+        final Gson gson = new Gson();
+        return gson.toJson(this);
+    }
+
+    static TokenCacheItem deserialize(String serializedBlob) throws AuthenticationException {
+        final Gson gson = new Gson();
+        try {
+            return gson.fromJson(serializedBlob, SSOStateContainer.class).getTokenItem();
+        } catch (final JsonParseException exception) {
```

Comment:
The goal here is to provide a method that hides the details of serialization.  If we are going to make a change I would prefer to go the other direction.  Either we rename this class to something like SSOStateSerializer and have serialize take a TokenCacheItem and deserialize return them, thus providing symmetry.  Or we wrape SSOStateContainer in another class and expse serialize and deserialize methods that deal with TokenCacheItems.

Either of those two implementations would hide the fact that we are wrapping the token in an outer JSON object.